### PR TITLE
Add User.set_note()

### PR DESCRIPTION
**Pull request by apple502j**
### Summary
This adds `await user.set_note("foo bar")`, a function which sets a "note" (shown in profile).

### Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by CapnS** [comment]
I don't know if it's worth the effort because this is only for userbots, but could you also  add User.fetch_note()?

**Comment by Soheab** [comment]
Ew selfbots.. but fetch note would cool? https://github.com/Rapptz/discord.py/pull/2254#issuecomment-507030695

**Comment by EEKIM10** [comment]
i dont really see a point in this. who even uses notes anyway?

**Comment by BluePhoenixGame** [comment]
I feel like discord.py should support the whole discord api. And that includes the notes!

**Comment by Phxntxm** [comment]
Personally...I actually *don't* want this included, I have already seen times where people think they can try to use notes as some kind of internal-to-discord database, which is dumb. I understand the reasoning behind leaving the already-included selfbot only, but I don't like adding another that I see people trying to abuse.

Maybe I have the abnormal opinion here, but just putting it out there.

File: discord/user.py [hunk 1]
```diff
@@ -849,3 +849,28 @@ def transform(d):
...
+
+        Sets the note.
+
+        .. note::
+
+            This only applies to non-bot accounts.
+
+        Parameters
+        -----------
+        note: :class:`str`
```
Comment (Rapptz) [inline discord/user.py]:
```suggestion
        note: Optional[:class:`str`]
```

File: discord/user.py [hunk 2]
```diff
@@ -849,3 +849,28 @@ def transform(d):
...
+
+        Raises
+        -------
+        Forbidden
+            Not allowed to set note.
+        HTTPException
+            Setting the note failed.
+
+        """
+        note = str(note) or ""
```
Comment (Rapptz) [inline discord/user.py]:
```suggestion
        note = "" if note is None else str(note)
```

File: discord/user.py [hunk 3]
```diff
@@ -849,3 +849,28 @@ def transform(d):
                        mutual_guilds=mutual_guilds,
                        user=self,
                        connected_accounts=data['connected_accounts'])
+
+    async def set_note(self, note):
```
Comment (Rapptz) [inline discord/user.py]: Honestly not sure I like this name.

Let me disregard the fact that this is a user-bot only feature for a minute.

In this library whenever you want to edit something about the model you don't use any `set_` methods, you have an `edit` method with a kwarg-only interface. Now, `User.edit` doesn't make much sense to add for only this thing but it's what would be the most consistent. The problem is that the concept of editing a `User` instance and adding an `edit` to it is that it breaks expectations when using `Member.edit`. Why couldn't you edit the note on a Member with the same interface?

These design decisions are why personally I'm not sure this is a good fit. Maybe if the name was something else. Maybe it's unimportant.

**Comment by bmintz** [comment]
I would rather have User.note which is None on bot accounts than User.fetch_note, unless this information is not dispatched along with the other fields of User.

**Comment by AtieP** [comment]
Why is this still not approved?

**Pull request closed without merging**